### PR TITLE
[TT-5524] Update bootstrap post-install hook and introduce new field in the values.yaml

### PR DESCRIPTION
**Pull request by buraksekili**
Current problem observed:
- `tyk-operator-conf` is not deleted if we do helm delete

- Failed Job doesn’t get deleted if we do helm delete. We end up with more jobs that fail

This PR adds 
- `before-hook-creation` to the delete policy of the post-install bootstrap. So that, jobs created by the bootstrap will be deleted either before a new hook is launched or successfully finished.
- new deletion hooks to clean up a secret of tyk-operator if it exists.
- new field called `operatorSecret` in the values.yaml. If it is set to true, `tyk-operator-conf` is created along with the dashboard bootstrap. Otherwise, it won't be created.

### CODE REVIEW

**Review by komalsukhani — APPROVED** [review]

**Comment by komalsukhani** [comment]
@buraksekili PR looks okay to me. I am not merging it yet since we plan to release it in next release.

**Review by andrei-tyk — APPROVED** [review]

**Review by komalsukhani — APPROVED** [review]

File: tyk-pro/templates/bootstrap-pre-delete.yaml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    # Removed the use of appVersion since it's not accurate. We'll put it back in v1.0
+    # when every app will have its own a chart
+    # app.kubernetes.io/version: {{ .Chart.AppVersion | quote }}
+
+    helm.sh/chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+  annotations:
+    # This is what defines this resource as a hook. Without this line, the
+    # job is considered part of the release.
+    "helm.sh/hook": pre-delete
+    "helm.sh/hook-weight": "-6"
```
Comment (komalsukhani) [inline tyk-pro/templates/bootstrap-pre-delete.yaml:19]: All looks good. Just curious to know why have to set weight to `-6`?

**Comment by komalsukhani** [comment]
@buraksekili Please resolve the conflicts and then we can merge it.

File: tyk-pro/templates/bootstrap-pre-delete.yaml (hunk 1 above)
Comment (buraksekili) [reply tyk-pro/templates/bootstrap-pre-delete.yaml:19]: My bad, I forgot to delete it. We can leave it with its default value since we have only one hook.

**Review by komalsukhani — APPROVED** [review]

**Pull request merged**